.github/workflows/nightly.yaml

```diff
@@ -22,7 +23,7 @@ jobs:
           pip install -e .[dev]
           pip install twine
       - name: Run tests
-        run: python -m unittest discover --verbose --start-directory . --pattern "*_test.py"
+        run: pytest --verbose -o python_files="*_test.py"
```

Comment:
The filter isn't necessary I don't think

Our main python tests just call `pytest`: https://github.com/pytorch/torchx/blob/9f20e5a55233288a3ba6fb9bec4e72d1e5fbf6c2/.github/workflows/python-unittests.yaml#L40